src/devices/CharacterLcd/LcdInterface.Gpio.cs

```diff
@@ -66,6 +67,7 @@ public Gpio(int registerSelectPin, int enablePin, int[] dataPins, int backlightP
                 }
 
                 _controller = controller ?? new GpioController(PinNumberingScheme.Logical);
+                _shouldDispose = shouldDispose;
```

Comment:
I would recommend to by default set this to false whenever the controller parameter wasn't null. So basically, if somebody passed in a controller, let's default shouldDispose to false.

Reply:
The current behavior was that everything was disposed if you passed one GpioController. So the shouldDispose is then by default true. 
Also this create consistency across all the bindings.

Reply:
I would prefer false as well, but it would be a breaking change for existing bindings. 
@krwq Your oppinion might be useful here.

Reply:
One thing we should definitely do is to always dispose when controller is null (or throw an exception when user passes null and asks for not disposing).

Considering our current guidelines is to always dispose and to not re-use GpioController IMO we should always dispose but I'm not strongly opinionated. I think it's much easier to explain what this method does if we consistently dispose or not dispose - if we choose to not dispose then we should change guidelines to match.

Reply:
OK, so let me adjust the code to:
- dispose whatever the shouldDispose parameter is if gpioController is null
- In case gpioController is not null, then follow the shouldDispose wish

Recommendatin is not to reuse GpioController but in some cases like when using FT4222 or a controller that can only be created once, you don't really have other options.